subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1868,7 +1874,7 @@ static ssize_t recvfrom_dtls_server(struct tls_context *ctx, void *buf,
 				if (ret == 0) {
 					repeat = true;
 				} else {
-					ret = -ECONNABORTED;
```

Comment:
The above function would fail if `mbedtls_ssl_session_reset()` fails, `ECONNABORTED` was used incorrectly as from what I've seen the function fails if it's not able to re-allocate resources from the heap, hence `ENOMEM`.